### ISSUE TITLE
Drop support for Python EOL versions

### ISSUE THREAD

**Issue by gaby** [ansible-community/ara#420]
## What is the idea ?
Looking at the documentation and code base, it looks like `ara` still supports or tries to support Python 3.5 and 3.6. These versions of Python are EOL since 2020 for 3.5 and 2021 for 3.6

Are there plans to address this?

Source: https://devguide.python.org/versions/

**Comment by gaby** [comment]
I can send a PR to update all the code to py3.7+, my only concern is that I can't figured out if the `zuul` tests are also testing the cli/server python code

**Comment by dmsimard** [comment]
Hi @gaby and thanks for the issue.

ara has not historically dropped support for python versions because they were EOL upstream with the rationale that there are plenty of long-term supported operating systems that still ship with these versions.
Distributions like Debian, Ubuntu LTS, CentOS, RHEL and the like.

The project dropped support for python versions when there was a need for it that couldn't be met otherwise and the trigger for the past couple of times when support was dropped was because we were upgrading the version of django which itself dropped support for something.

In the [last release notes](https://github.com/ansible-community/ara/releases/tag/1.5.8) there was a note about python support:
```
- ara 1.5.8 is the last release that will support python3.5.
  Python 3.5 reached the end of its life on September 13th, 2020.
  An upcoming release will update the version of django to the next LTS (2.2 to 3.2)
  which will bump the requirement to python>=3.6.
```

So, in other words, the next release of ara (1.6.0) will raise the requirement to py36 (because of django 3.2) but there is no requirement on higher for the moment.

What did you have in mind about py37+ ? We can consider doing it in the future but not right now.

**Comment by gaby** [comment]
Makes sense to me to move the base to py36 at least. Specially since OS's like Ubuntu 18.04 LTS still use that version.

Using `pyupgrade` it's very easy to reformat all the code-base to any version of python and forward: https://github.com/asottile/pyupgrade

**Comment by gaby** [comment]
I just ran it in a local copy of `ara` and it found changes in all these files:

![image](https://user-images.githubusercontent.com/835733/189501710-7e5530de-f47a-4849-86ae-a2a90616e761.png)

Example changes:

![image](https://user-images.githubusercontent.com/835733/189501722-ad4956e6-7500-4683-ae92-ac8011fafe7b.png)

**Comment by dmsimard** [comment]
Oh, nice, I didn't know about pyupgrade.

Feel free to open a PR for updating to 3.6 (keeping ``%s`` is fine too), we'll double check if it causes any regressions but it looks OK at first glance.

**Comment by gaby** [comment]
> Oh, nice, I didn't know about pyupgrade.
> 
> Feel free to open a PR for updating to 3.6 (keeping `%s` is fine too), we'll double check if it causes any regressions but it looks OK at first glance.

PR submitted
